rocksdb_admin/admin_handler.cpp

```diff
@@ -249,6 +249,69 @@ std::unique_ptr<rocksdb::DB> AdminHandler::removeDB(
   return db;
 }
 
+void AdminHandler::async_tm_addDB(
+      std::unique_ptr<apache::thrift::HandlerCallback<std::unique_ptr<
+          AddDBResponse>>> callback,
+      std::unique_ptr<AddDBRequest> request) {
+  db_admin_lock_.Lock(request->db_name);
+  SCOPE_EXIT { db_admin_lock_.Unlock(request->db_name); };
+
+  AdminException e;
+  // validate the request
```

Comment:
Wouldn't you consider adding an existing DB an invalid request? But sure I will remove it